src/linker/Linker/MessageContainer.cs

```diff
@@ -74,14 +64,24 @@ public static MessageContainer CreateErrorMessage (string text, int code, string
 		/// <param name="version">Optional warning version number. Versioned warnings can be controlled with the
 		/// warning wave option --warn VERSION. Unversioned warnings are unaffected by this option. </param>
 		/// <returns>New MessageContainer of 'Warning' category</returns>
-		public static MessageContainer CreateWarningMessage (LinkContext context, string text, int code, MessageOrigin origin, string subcategory = MessageSubCategory.None, WarnVersion? version = null)
+		public static MessageContainer CreateWarningMessage (LinkContext context, string text, int code, MessageOrigin origin, WarnVersion version, string subcategory = MessageSubCategory.None)
 		{
-			ValidateWarning (code, version);
+			if (!(code > 2000 && code <= 6000))
```

Comment:
Not sure if you wanted to fix it here, but this doesn't address https://github.com/mono/linker/issues/1384.

Reply:
I'm choosing to specifically not address #1384 with this PR - it's related to the intent of these changes, but not really. We need this PR to fix the --nowarn --warnaserror problems, so focusing on that part.